Lib/inspect.py

```diff
@@ -1672,10 +1672,34 @@ def getframeinfo(frame, context=1):
 
     return Traceback(filename, lineno, frame.f_code.co_name, lines, index)
 
+
+def _tblineno(tb):
+    tb_lineno = tb.fb_lineno
+    if tb_lineno is not None:
+        return tb_lineno
+
+    return _lasti2lineno(tb.tb_frame.f_code, tb.tb_lasti)
+
+
+def _lasti2lineno(code, lasti):
```

Comment:
there's a lasti2lineno implementation here - but it uses a reversed list(dis.finelinestarts(code)) 
https://github.com/python/cpython/blob/deeaac49e267285158264643799624623f4a7b29/Lib/pdb.py#L119-L126